### PR TITLE
CA-314170: wait for frontend_closed with qemu-dp too

### PR DESCRIPTION
**Pull request by edwintorok**
Done some initial testing and unplug worked, will have to rerun the original failing test and confirm that it passes now.

### CODE REVIEW

File: xc/device.ml [hunk 1]
```diff
@@ -212,20 +212,27 @@ module Generic = struct
...
     if not qdisk then begin
       Watch.key_to_disappear path
     end else begin
-      debug "unplug_watch: not waiting for qdisk %s; returning dummy watch" path;
-      Watch.{ evaluate = fun xs -> try xs.Xs.rm path with _ -> debug "dummy unplug_watch for qdisk: %s already removed" path }
+      (* We need to delete this key ourselves, qemu-dp wouldn't delete it *)
+      debug "unplug_watch: not waiting for qdisk %s; waiting for frontend closed" path;
+      (* Wait until either we've already deleted the key, or the frontend has reached the closed
+       * state *)
+      Watch.any_of [
```
Comment (lindig) [inline xc/device.ml]: I have trouble parsing this code. Could the formatting be improved to highlight the structure?

Comment (edwintorok) [reply xc/device.ml]: I've added another variable to make it clearer what goes into `Watch.any_of` (it takes 2 other watches, now each one has an explicit name).

**Comment by lindig** [comment]
I'd like to see a Travis run before we merge this. One commit seems to miss a sign-off according to the test.

**Comment by edwintorok** [comment]
I think Travis didn't run because this was a draft PR, converted to regular one now.

**Comment by edwintorok** [comment]
Closing this for now, lots of failures in tests with Watch.Timeout

**Pull request closed without merging**